2-mandatory/6-journey-planner.js

```diff
@@ -123,6 +129,8 @@ function getLocationName() {}
 */
 function journeyPlanner(locations, transportMode) {
   // Implement the function body
+  let result = locations.filter((place) => place.includes(transportMode));
+  return result.map((location) => location[0]);
 }
 
```

Comment:
I used reduce to solve this, but you definitely have the knack of writing short efficient code. Here's mine:
```javascript
function journeyPlanner(locations, transportMode) {
  return locations.reduce(
    (reachableLocations, currentLocationWithTransports) => {
      const [location, ...transportsForLocation] = currentLocationWithTransports;
      if (transportsForLocation.find(mode => mode === transportMode)) {
        return [...reachableLocations, location];
      } else {
        return reachableLocations;
      }
    },
    []
  );
}
```
Mine does the job and is quite fancy, but yours is simpler and a lot more readable than mine, so well done!